engine/tests/ElggUserTest.php

```diff
@@ -131,7 +131,9 @@ public function testElggUserNameCache() {
 		$guid = $this->user->save();
 
 		$user = get_user_by_username($name);
-		$user->delete();
+
+		$this->assertTrue($user->delete());
```

Comment:
There was a bug in `ElggEntity::delete()`. Cache was invalidated, but it was being repopulated in one of the consequent delete operations (not sure which one - metadata, relationships or other). I have moved cache invalidation closer to the DB delete query.